implementations/scripts/Commit.py

```diff
@@ -0,0 +1,90 @@
+import math
+import pandas as pd
+from Metric import Metric
+import utils
+
+
+class Commit(Metric):
+    """
+    Initilizes self.df, the dataframe with one commit per row.
+
+    :param data_list: A list of dictionaries.
+        Each element is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+
+    :param date_range: A tuple which represents the period of interest
+        It is of the form (since, until), where since and until are strings.
+        Either, or both can be None. If, for example, since is None, that 
+        would mean that all commits from the first commit to the commit who last
+        falls inside the until range will be included.
+
+    :param is_src_code_obj: An object of IsSourceCode.
+        It is used to determine what comprises source code.
+        """
+
+    def __init__(self, data_list, date_range=(None, None), is_src_code_obj=None):
+
+        super().__init__(data_list)
+
+        clean_data_list = list()
+        (self.since, self.until) = date_range
+
+        for line in self.raw_df.iterrows():
+            commit = line[1].to_dict()
+            commit = self._clean_commit(commit)
+            if is_src_code_obj is None  \
+                    or is_src_code_obj.is_source_code(commit):
+                clean_data_list.append(commit)
+
+        self.df = pd.DataFrame(clean_data_list)
+        if self.since:
+            self.df = self.df[self.df['created_date']
+                              >= utils.str_to_dt_other(self.since)]
```

Comment:
Since we have datetime in Python for dates, we should just use it for since / until, and for dates coming from Perceval.

In other words, let's make all our classes work with datetime, so that people can use the usual Python conversions (if needed) when using them. For that, the functioning flattening data from Perceval would be in charge of converting dates, as produced by Perceval, to datetime.

Reply:
Got it!